### PR TITLE
Add basic UI

### PR DESCRIPTION
**Pull request by Basir-Mohammadi**
## Project requirements
- Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person).
- Create a book.
- Create a rental.
- List all rentals for a given person id.
- In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
- Present the user with a list of options to perform.
- Lets users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+require './student'
+require './teacher'
+require './book'
+require './classroom'
+
+class App
+  attr_reader :books, :classrooms, :students, :teachers
+
+  def initialize
```
Comment (rustam-tolipov) [inline app.rb:9]: You made a great effort to create this project. However, some changes need to be done.

File: factory.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    options = [
+      'Create person',
+      'Create classroom',
+      'Create book',
+      'Create rental',
+      'List all books',
+      'List all people',
+      'List all classrooms',
+      'List all rentals by person',
+      'Quit'
```
Comment (rustam-tolipov) [inline factory.rb:41]: - [ ] I'm having some problems with what to create first. Also, you are not required for creating a classroom here. But it's a good idea. But when I try to create the person and see it, it's breaking the app. Kindly watch this video and fix the error. Make sure there are no errors while we run your app. 

https://user-images.githubusercontent.com/69011963/189472656-8ffee115-f885-4bd4-ac95-ebb7e6e9e9c7.mov

File: factory.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    ]
+    list_with_index(options)
+  end
+end
```
Comment (rustam-tolipov) [inline factory.rb:45]: [OPTIONAL] Kindly add MIT license to your project.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Basir-Mohammadi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    student = Student.new(age, name, classroom)
+    @students << student
+    student
+  end
+
+  def create_teacher(age, name, specialization)
+    teacher = Teacher.new(age, name, specialization)
+    @teachers << teacher
+    teacher
+  end
```
Comment (cynthiainga) [inline app.rb:40]: - [x] Please as mentioned by the previous reviewer, you should make sure that your app works correctly. When I try to create a person it's breaking the app, please kindly fix the error.
You can take a look at how the app breaks when I create a person 🙃 

https://user-images.githubusercontent.com/74501523/189474365-8dae4bc2-7385-4235-bc7f-9965f9d3239f.mov

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Basir-Mohammadi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.



 ## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ahmetbozaci — APPROVED** [review]
# 🥳 🎉🥇 💯  💥 🎆 🎈 STATUS: APPROVED 🥳 🎉  🥇  💯  💥   🎆 🎈 🎉 



@Basir-Mohammadi  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.
___ 

### 💯 Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

aproved

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**